core/capabilities/ccip/configs/solana/chain_writer.go

```diff
@@ -0,0 +1,330 @@
+				},
+				IsSigner:   false,
+				IsWritable: false,
+			},
+			chainwriter.AccountLookup{
+				Name:     "UserAccounts",
+				Location: "Message.ExtraArgs.Accounts",
+			},
+			chainwriter.PDALookups{
+				Name: "ReceiverAssociatedTokenAccount",
+				PublicKey: chainwriter.AccountConstant{
+					Address: solana.SPLAssociatedTokenAccountProgramID.String(),
+				},
+				Seeds: []chainwriter.Seed{
+					{Dynamic: chainwriter.AccountLookup{Location: "Message.Receiver"}},
+					{Dynamic: chainwriter.AccountsFromLookupTable{
+						LookupTableName: "RegistryTokenState",
+						IncludeIndexes:  []int{5},
+					}},
+					{Dynamic: chainwriter.AccountLookup{Location: destTokenAddress}},
+				},
+			},
+			chainwriter.PDALookups{
+				Name: "PerChainTokenConfig",
+				PublicKey: chainwriter.AccountConstant{
+					Address: routerProgramAddress,
+				},
+				Seeds: []chainwriter.Seed{
+					{Dynamic: chainwriter.AccountLookup{Location: destTokenAddress}},
+					{Dynamic: chainwriter.AccountLookup{Location: destChainSelectorPath}},
+				},
+				IsSigner:   false,
+				IsWritable: false,
+			},
+			chainwriter.AccountsFromLookupTable{
+				LookupTableName: "RegistryTokenState",
+				IncludeIndexes:  []int{},
+			},
+			chainwriter.PDALookups{
+				Name: "RegistryTokenConfig",
+				PublicKey: chainwriter.AccountConstant{
+					Address:    routerProgramAddress,
+					IsSigner:   false,
+					IsWritable: false,
+				},
+				Seeds: []chainwriter.Seed{
+					{Dynamic: chainwriter.AccountLookup{Location: destTokenAddress}},
+				},
+				IsSigner:   false,
+				IsWritable: false,
+			},
+			chainwriter.PDALookups{
+				Name: "UserNoncePerChain",
+				PublicKey: chainwriter.AccountConstant{
+					Address:    routerProgramAddress,
+					IsSigner:   false,
+					IsWritable: false,
+				},
+				Seeds: []chainwriter.Seed{
+					{Dynamic: chainwriter.AccountLookup{Location: "Message.Receiver"}},
+					{Dynamic: chainwriter.AccountLookup{Location: destChainSelectorPath}},
+				},
+			},
+			chainwriter.PDALookups{
+				Name: "CPISigner",
+				PublicKey: chainwriter.AccountConstant{
+					Address:    routerProgramAddress,
+					IsSigner:   false,
+					IsWritable: false,
+				},
+				Seeds: []chainwriter.Seed{
+					{Static: []byte("external_token_pools_signer")},
+				},
+				IsSigner:   false,
+				IsWritable: false,
+			},
+			chainwriter.AccountConstant{
+				Name:       "ComputeBudgetProgram",
+				Address:    computeBudgetProgramAddress,
+				IsSigner:   true,
+				IsWritable: false,
+			},
+		},
+		DebugIDLocation: "Message.MessageID",
+	}
+}
+
+func GetSolanaChainWriterConfig(routerProgramAddress string, commonAddressesLookupTable solana.PublicKey, fromAddress string) (chainwriter.ChainWriterConfig, error) {
+	computeBudgetProgramAddress := solana.ComputeBudget.String()
+	sysvarInstructionsAddress := solana.SysVarInstructionsPubkey.String()
+
+	// check fromAddress
+	_, err := solana.PublicKeyFromBase58(fromAddress)
```

Comment:
Should we do any other validations on the `fromAddress` like `IsZero()` or `IsOnCurve()` for sanity? IIUC the address correlates to a deployed program so it shouldn't be zero and should be on the curve